### ISSUE TITLE
reproject_exact() fails with inverted (swapped) WCS

### ISSUE THREAD

**Issue by evertrol** [astropy/reproject#113]
When I use `reproject_exact()` to project an image with a N-W direction onto an image with a N-E direction, I get incorrect values. `reproject_interp()` appears to do the correct thing.

Am I using the WCS incorrectly, or am I overlooking something else?

Below is a test-case. Other than one column of NaNs, which I'm happy to accept as the result of 1--0 FITS/Python offsets, the interpolated version gives me the correct answer. 
(The array before divided by its weights in `_reproject_celestial` indeed contains only 0's, hence the NaNs in the final output.)

``` python
import numpy as np
from astropy.io.fits import PrimaryHDU
from astropy.wcs import WCS
from reproject import reproject_interp, reproject_exact

N = 5

wcs = WCS(naxis=2)
wcs.wcs.ctype = ["RA---TAN", "DEC--TAN"]
wcs.wcs.crval = [0, 0]
wcs.wcs.crpix = [N/2, N/2]
wcs.wcs.cdelt = [-0.01, 0.01]
header = wcs.to_header()
container = PrimaryHDU(data=np.zeros((N, N)), header=header)
print(WCS(header))

wcs = WCS(naxis=2)
wcs.wcs.ctype = ["RA---TAN", "DEC--TAN"]
wcs.wcs.crval = [0, 0]
wcs.wcs.crpix = [N/2, N/2]
wcs.wcs.cdelt = [0.01, 0.01]
header = wcs.to_header()
small = PrimaryHDU(data=np.outer(np.ones(N), np.arange(1, N+1)), header=header)
print(WCS(header))
print('small:')
print(small.data)
print()

array, footprint = reproject_exact(small, container.header)
print('exact:')
print('array:')
print(array)
print('footprint:')
print(footprint)
print(10*'- ')

array, footprint = reproject_interp(small, container.header)
print('interpolated:')
print('array:')
print(array)
print('footprint:')
print(footprint)
```

which yields

``` python
WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---TAN'  'DEC--TAN'  
CRVAL : 0.0  0.0  
CRPIX : 2.5  2.5  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : -0.01  0.01  
NAXIS    : 0 0
WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---TAN'  'DEC--TAN'  
CRVAL : 0.0  0.0  
CRPIX : 2.5  2.5  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : 0.01  0.01  
NAXIS    : 0 0
small:
[[ 1.  2.  3.  4.  5.]
 [ 1.  2.  3.  4.  5.]
 [ 1.  2.  3.  4.  5.]
 [ 1.  2.  3.  4.  5.]
 [ 1.  2.  3.  4.  5.]]

exact:
array:
[[ nan  nan  nan  nan  nan]
 [ nan  nan  nan  nan  nan]
 [ nan  nan  nan  nan  nan]
 [ nan  nan  nan  nan  nan]
 [ nan  nan  nan  nan  nan]]
footprint:
[[ 0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.]
 [ 0.  0.  0.  0.  0.]]
- - - - - - - - - - 
interpolated:
array:
[[  4.   3.   2.   1.  nan]
 [  4.   3.   2.   1.  nan]
 [  4.   3.   2.   1.  nan]
 [  4.   3.   2.   1.  nan]
 [  4.   3.   2.   1.  nan]]
footprint:
[[ 1.  1.  1.  1.  0.]
 [ 1.  1.  1.  1.  0.]
 [ 1.  1.  1.  1.  0.]
 [ 1.  1.  1.  1.  0.]
 [ 1.  1.  1.  1.  0.]]
```

(I've played around with different ways to create the second WCS, using `wcs.wcs.pc` and `wcs.wcs.cd` matrices, but to no avail. The fact that the interpolation variant can handle this and the image displays fine in eg DS9, suggests to me that this is not the issue.)

**Comment by evertrol** [comment]
It appears `computerOverlap` doesn't handle the overlap between clockwise and counter-clockwise cells properly. Possibly, this is because the outer product changes sign, but I haven't investigated that.

Here's a test case:

```c
/*
 * Compile with:
 *     cc -Wall -Wpedantic -I reproject/spherical_intersect test_overlap2.c reproject/spherica\
l_intersect/overlapArea.c -lm -o test_overlap2
 */

#include <stdlib.h>
#include <stdio.h>
#include <float.h>
#include <math.h>
#include "overlapArea.h"
                                                                       
int
main()
{
        // i: counter-clockwise
        double ilon[4] = {0.1, 0.2, 0.2, 0.1};
        double ilat[4] = {0.1, 0.1, 0.2, 0.2};
        // o: clockwise
        double olon[4] = {0.1, 0.1, 0.2, 0.2};
        double olat[4] = {0.1, 0.2, 0.2, 0.1};
        double ratio = 0;

        double overlap = computeOverlap(ilon, ilat, olon, olat, 0, 1, &ratio);
        printf("overlap = %13.6e\n", overlap);

        // change o to be counter-clockwise                                                    
        double tmp = olon[0];
        olon[0] = olon[2];
        olon[2] = tmp;
        tmp = olat[0];
        olat[0] = olat[2];
        olat[2] = tmp;
        overlap = computeOverlap(ilon, ilat, olon, olat, 0, 1, &ratio);
        printf("overlap = %13.6e\n", overlap);

        return EXIT_SUCCESS;
}
```                                                                                               

The output I get here is 
```bash
overlap =  0.000000e+00
overlap =  9.891264e-03
```

**Comment by astrofrog** [comment]
@evertrol - this issue is now fixed in reproject v0.5.1
